tests/test_epic_dist.py

```diff
@@ -0,0 +1,50 @@
+import gym
+import numpy as np
+import pytest
+
+from epic.distances import epic
+from epic.samplers import DummyGymStateSampler, GymSpaceSampler
+
+
+def rew_fn_1(state, action, next_state, _):
+    return state + next_state + action
+
+
+def rew_fn_2(state, action, next_state, _):
+    return state**2 + next_state**2 + 2 * action
+
+
+def rew_fn_1_potential_shaping(state, action, next_state, _):
```

Comment:
Might end up being useful to have a wrapper that takes a `states -> floats` function and returns a `RewardFunction`, then we could easily test things like this for a bunch of combinations of reward functions and potentials. (Not super high priority)

If the wrapper was written such that it also supported pytorch tensors and backprop, I think it could also be used for the implementation of div-free rewards (where we have a neural net describing the `states -> floats` function, but care about the associated reward function for canonicalization)

Reply:
Just to make sure I'm following - this just adds the `states -> floats` function as a potential to a `RewardFunction`?

I think implementing this wrapper and making it support backprop seems good. I could also potentially take this on when I implement the div-free reward distance calculation.

Reply:
Yeah exactly, that's the wrapper I had in mind. The only issue with supporting backprop (AFAICT) would be that the `RewardFunction` type currently requires numpy arrays. I'd guess we don't just want to generalize `RewardFunction` to allow pytorch tensors because presumably not all of the code supports pytorch (though I haven't checked). Maybe just have a more general type signature specifically for that wrapper

Reply:
Oh, to be clear: I think the wrapper doesn't necessarily _add_ to a reward function that it takes as input---it could just return the part that would need to be added, i.e. gamma * Phi(s') - Phi(s), which is itself a valid reward function. If needed, this "pure potential shaping reward function" could then be added to some other existing reward function outside the wrapper. Though tbf, in most use cases we probably _do_ want to add this pure potential shaping term to a specific reward function, so writing the wrapper that way also seems reasonable